Add the network callback for wifi connection check

kaishi05 opened the pull request:
Use the network callback for wifi connection status check for L+ devices.

Also remove networkId check for pre-L devices as it doesn't sometimes work while the network is already connected.

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/google/mobly-bundled-snippets/195)
<!-- Reviewable:end -->

xpconanfan reviewed (changes requested):
These RPCs are not specific to wifi network right?
Seem like they belong to the network snippet?

kaishi05 commented:
Removed the internet check and only check if the wifi is available using networkCallback for L+ devices.. Removed networkId check in Pre-L devices.

Note that it is recommended by the framework to use networkCallback to check Wifi status. So this file is the right place to add the networkCallback to check Wifi status.

kaishi05 commented:
Please  review 2nd comment.

mhaoli commented on src/main/java/com/google/android/mobly/snippet/bundled/WifiManagerSnippet.java:
```diff
@@ -373,9 +414,9 @@ public void wifiEnableSoftAp(@Nullable JSONObject configuration) throws Throwabl
-                            + configuration);
+      throw new WifiManagerSnippetException(
+          "Timed out after 60s waiting for Wi-Fi Soft AP state to turn on with configuration: "
+              + configuration);
```
Could we revert the indent change in this PR? The modified lines look inconsistent with unmodified lines.

kaishi05 replied:
revert the unnecessary indent changes.

mhaoli reviewed (approved):
LGTM as this is the recommended WiFi status check approach from WiFi framework.

And this approach works for both WiFi network w/ and w/o internet access.
